### PR TITLE
Fixes #27265 - Improve domains api test

### PR DESCRIPTION
**Pull request by tbrisker**
Fix transient failure and refactor taxonomy scope testing.

### CODE REVIEW

**Comment by ekohl** [comment]
[test katello]

Odd failure
```
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "http_proxies" does not exist
LINE 8:                WHERE a.attrelid = '"http_proxies"'::regclass
                                          ^
:               SELECT a.attname, format_type(a.atttypid, a.atttypmod),
                     pg_get_expr(d.adbin, d.adrelid), a.attnotnull, a.atttypid, a.atttypmod,
                     c.collname, col_description(a.attrelid, a.attnum) AS comment
                FROM pg_attribute a
                LEFT JOIN pg_attrdef d ON a.attrelid = d.adrelid AND a.attnum = d.adnum
                LEFT JOIN pg_type t ON a.atttypid = t.oid
                LEFT JOIN pg_collation c ON a.attcollation = c.oid AND a.attcollation <> t.typcollation
               WHERE a.attrelid = '"http_proxies"'::regclass
                 AND a.attnum > 0 AND NOT a.attisdropped
               ORDER BY a.attnum
```

**Comment by jlsherrill** [comment]
[test katello]

File: test/controllers/api/v2/domains_controller_test.rb [hunk 1]
```diff
@@ -64,42 +64,42 @@ class Api::V2::DomainsControllerTest < ActionController::TestCase
...
+    let (:mydomain) { domains(:mydomain) }
+    let (:yourdomain) { domains(:yourdomain) }
+    let (:loc) { FactoryBot.create(:location, domains: [mydomain, yourdomain]) }
+    let (:org) { FactoryBot.create(:organization, domains: [mydomain]) }
+
+    test "should get domains for location only" do
+      get :index, params: { :location_id => loc.id }
+      assert_response :success
+      assert_equal loc.domains.length, assigns(:domains).length
+      assert_equal assigns(:domains).sort, loc.domains.sort
```
Comment (timogoebel) [inline test/controllers/api/v2/domains_controller_test.rb]: I believe assert_same_elements could help here.

Comment (tbrisker) [reply test/controllers/api/v2/domains_controller_test.rb]: nice! didn't know that one.

File: test/controllers/api/v2/domains_controller_test.rb [hunk 2]
```diff
@@ -64,42 +64,42 @@ class Api::V2::DomainsControllerTest < ActionController::TestCase
...
-    get :index, params: { :organization_id => taxonomies(:organization1).id }
-    assert_response :success
-    assert_equal taxonomies(:organization1).domains.length, assigns(:domains).length
-    assert_equal taxonomies(:organization1).domains, assigns(:domains)
-  end
+    test "should get domains for organization only" do
+      get :index, params: { :organization_id => org.id }
+      assert_response :success
+      assert_equal org.domains.length, assigns(:domains).length
+      assert_equal org.domains, assigns(:domains)
```
Comment (ezr-ondrej) [inline test/controllers/api/v2/domains_controller_test.rb]: wouldn't be better to be consistent and use `assert_same_elements` here as well (I know it is not very usefull, but just to be consistent)?

File: test/controllers/api/v2/domains_controller_test.rb [hunk 3]
```diff
@@ -64,42 +64,42 @@ class Api::V2::DomainsControllerTest < ActionController::TestCase
...
-    get :index, params: { :location_id => taxonomies(:location1).id, :organization_id => taxonomies(:organization1).id }
-    assert_response :success
-    assert_equal 1, assigns(:domains).length
-    assert_equal assigns(:domains), [domains(:mydomain)]
-  end
+    test "should get domains for both location and organization" do
+      get :index, params: { :location_id => loc.id, :organization_id => org.id }
+      assert_response :success
+      assert_equal 1, assigns(:domains).length
+      assert_equal assigns(:domains), [mydomain]
```
Comment (ezr-ondrej) [inline test/controllers/api/v2/domains_controller_test.rb:91]: Same in here (`assert_same_elements`)

**Review by ezr-ondrej — COMMENTED** [review]
Perfect @tbrisker! Just a followup nitpick on `same_element`, otherwise LGTM.

File: test/controllers/api/v2/domains_controller_test.rb (hunk 2 above)
Comment (tbrisker) [reply test/controllers/api/v2/domains_controller_test.rb]: changed here

File: test/controllers/api/v2/domains_controller_test.rb (hunk 3 above)
Comment (tbrisker) [reply test/controllers/api/v2/domains_controller_test.rb:91]: but not here to emphasize this should only be the one domain and not both

Comment (ezr-ondrej) [reply test/controllers/api/v2/domains_controller_test.rb:91]: Got it, I am ok with that :)

**Review by ezr-ondrej — APPROVED** [review]
Thanks @tbrisker, works locally, and seems it should fix the tests :+1:
Lets test it out, merging!

**Pull request merged**